tests/test_examples_sklearnex.py

```diff
@@ -36,40 +38,31 @@
 class TestsklearnexExamples(unittest.TestCase):
     """Class for testing sklernex examples"""
 
-    # Get a list of all Python files in the examples directory
-    pass
-
-
-def test_generator(file):
-    def testit(self):
-        # Run the script and capture its exit code
-        process = subprocess.run(
-            [sys.executable, os.path.join(examples_path, file)],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            check=False,
-        )  # nosec
-        exit_code = process.returncode
-
-        # Assert that the exit code is 0
-        self.assertEqual(
-            exit_code,
-            0,
-            msg=f"Example has failed, the example's output:\n{process.stdout.decode()}\n{process.stderr.decode()}",
-        )
-
-    setattr(TestsklearnexExamples, "test_" + os.path.splitext(file)[0], testit)
-    print("Generating tests for " + os.path.splitext(file)[0])
-
-
-files = [
-    f
-    for f in os.listdir(examples_path)
-    if f.endswith(".py") and "spmd" not in f and "dpnp" not in f and "dpctl" not in f
-]
-
-for file in files:
-    test_generator(file)
+    @pytest.mark.parametrize(
+        "file",
+        [
+            f
+            for f in os.listdir(examples_path)
+            if f.endswith(".py")
+            and "spmd" not in f
+            and "dpnp" not in f
+            and "dpctl" not in f
+        ],
+    )
+    def test_generator(file):
+        def testit(self):
```

Comment:
Dumb question, why is the ```testit``` function necessary now since we are no longer setting attributes to the class ```TestsklearnexExamples``` (is the class necessary anymore?)

Reply:
Thanks for pointing this out. I just realized that `pytest.parametrize` doesn't actually work when put into methods that inherit from `unittest.TestCase`, so the parametrizations where not actually being executed. I changed it now to regular `pytest`.